src/Compilers/CSharp/Test/Semantic/Semantics/PropertyFieldKeywordTests.cs

```diff
@@ -1292,6 +1292,299 @@ public int P
+            comp.TestOnlyCompilationData = accessorBindingData;
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers().OfType<FieldSymbol>());
+            comp.VerifyDiagnostics(
+                // (10,39): error CS8422: A static local function cannot contain a reference to 'this' or 'base'.
+                //             static int localFunc() => field;
+                Diagnostic(ErrorCode.ERR_StaticLocalFunctionCannotCaptureThis, "field").WithLocation(10, 39)
+            );
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void InLocalFunction()
+        {
+            var comp = CreateCompilation(@"
+public class C
+{
+    public int P
+    {
+        get
+        {
+            return localFunc();
+            int localFunc() => field;
+        }
+    }
+}
+");
+            var accessorBindingData = new SourcePropertySymbolBase.AccessorBindingData();
+            comp.TestOnlyCompilationData = accessorBindingData;
+            Assert.Empty(comp.GetTypeByMetadataName("C").GetMembers().OfType<FieldSymbol>());
+            comp.VerifyDiagnostics();
+            VerifyTypeIL(comp, "C", @"
+.class public auto ansi beforefieldinit C
+	extends [mscorlib]System.Object
+{
+	// Fields
+	.field private initonly int32 '<P>k__BackingField'
+	.custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+		01 00 00 00
+	)
+	// Methods
+	.method public hidebysig specialname 
+		instance int32 get_P () cil managed 
+	{
+		// Method begins at RVA 0x2050
+		// Code size 7 (0x7)
+		.maxstack 8
+		IL_0000: ldarg.0
+		IL_0001: call instance int32 C::'<get_P>g__localFunc|1_0'()
+		IL_0006: ret
+	} // end of method C::get_P
+	.method public hidebysig specialname rtspecialname 
+		instance void .ctor () cil managed 
+	{
+		// Method begins at RVA 0x2058
+		// Code size 7 (0x7)
+		.maxstack 8
+		IL_0000: ldarg.0
+		IL_0001: call instance void [mscorlib]System.Object::.ctor()
+		IL_0006: ret
+	} // end of method C::.ctor
+	.method private hidebysig 
+		instance int32 '<get_P>g__localFunc|1_0' () cil managed 
+	{
+		.custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerGeneratedAttribute::.ctor() = (
+			01 00 00 00
+		)
+		// Method begins at RVA 0x2060
+		// Code size 7 (0x7)
+		.maxstack 8
+		IL_0000: ldarg.0
+		IL_0001: ldfld int32 C::'<P>k__BackingField'
+		IL_0006: ret
+	} // end of method C::'<get_P>g__localFunc|1_0'
+	// Properties
+	.property instance int32 P()
+	{
+		.get instance int32 C::get_P()
+	}
+} // end of class C
+");
+            Assert.Equal(0, accessorBindingData.NumberOfPerformedAccessorBinding);
+        }
+
+        [Fact]
+        public void InStaticLambda_PropertyIsStatic()
+        {
+            var comp = CreateCompilation(@"
+using System;
+public class C
+{
+    public static int P
+    {
+        get
+        {
+            Func<int> f = static () => field;
```

Comment:
Honestly, this one feels a little odd to me. It's a `static` lambda, and while the `field` backing keyword is effectively a static field, it doesn't feel right to me that a lambda (or local function) marked static can affect the codegen of the containing member in such a way. Let's bring this as a question for LDM.

Reply:
> it doesn't feel right to me that a lambda (or local function) marked static can affect the codegen of the containing member in such a way.

@333fred I don't see how the lambda affected the codegen? the property itself is declared static already. See `InStaticLambda` for the test case where the property is not static which is an error case.

Reply:
> Yes, because if there is no reference to `field` it's now a regular property instead of a semi auto property? It's unclear to me why the behavior is questionable/suspicious. We can discuss in the next PR since I'll be reverting this part of the PR per your feedback.

A `static` lambda or local function is supposed to be something I can look at without needing to understand the context of the surrounding member. It being able to affect the context of the surrounding member worries me.

Reply:
@333fred That would be a question of whether to support `field` in local functions vs not supporting it in the first place right?

Reply:
No, I mean should `static` local functions/lambdas be able to access `field` at all.